### PR TITLE
Csp 121 course management addition

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.Roatp.CourseManagement.Application/Providers/Queries/GetProviderQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        }
+        
+        public async Task<GetProviderQueryResult> Handle(GetProviderQuery request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Get Provider request received for Ukprn number {ukprn}", request.Ukprn);
+            var provider = await _apiClient.Get<Domain.ApiModels.Provider>($"providers/{request.Ukprn}");
+            if (provider == null)
+            {
+                _logger.LogError("Provider not found for {ukprn}", request.Ukprn);
+                throw new ValidationException($"Provider not found for UKPRN {request.Ukprn}",null);
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Application/Providers/Queries/GetProviderQueryHandler.cs:27]: This should be InvalidOperationException

File: src/SFA.DAS.Roatp.CourseManagement.Web/Infrastructure/RouteNames.cs [hunk 2]
```diff
@@ -39,8 +40,8 @@ public static class RouteNames
 
         public const string GetAddProviderLocationDetails = "GetAddProviderLocationDetails";
         public const string PostAddProviderLocationDetails = "PostAddProviderLocationDetails";
-
         public const string GetUpdateProviderLocationDetails = "GetUpdateProviderLocationDetails";
         public const string PostUpdateProviderLocationDetails = "PostUpdateProviderLocationDetails";
+        public const string ProviderDescription = "provider-description";
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Infrastructure/RouteNames.cs:45]: This should follow the convention `GetProviderDescription`

File: src/SFA.DAS.Roatp.CourseManagement.Domain/ApiModels/Provider.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        public int Ukprn { get; set; }
+        public string LegalName { get; set; }
+        public string TradingName { get; set; }
+        public string Email { get; set; }
+        public string Phone { get; set; }
+        public string Website { get; set; }
+        public string MarketingInfo { get; set; }
+        public decimal? EmployerSatisfaction { get; set; }
+        public decimal? LearnerSatisfaction { get; set; }
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Domain/ApiModels/Provider.cs:13]: We should not be exposing feedback data out of outer api. This is not a concern for web and should be filtered out. Actually we should filter these out at inner api level.

**Review by DevenKShah — CHANGES_REQUESTED** [review]
Feedback fields should not be exposed from inner api. These are not required for now.

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ReviewYourDetailsControllerTests.cs [hunk 4]
```diff
@@ -44,11 +45,22 @@ public void Index_ReturnsViewWithModel()
                {
                    verifyRouteValues = c;
                });
+
+            urlHelper
+                .Setup(m => m.RouteUrl(It.Is<UrlRouteContext>(c =>
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ReviewYourDetailsControllerTests.cs:50]: I don't understand this request

[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ReviewYourDetailsControllerTests.cs]: Ok all done!

File: src/SFA.DAS.Roatp.CourseManagement.Application/Providers/Queries/GetProviderQueryHandler.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement.Application/Providers/Queries/GetProviderQueryHandler.cs]: fixed

File: src/SFA.DAS.Roatp.CourseManagement.Web/Infrastructure/RouteNames.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement.Web/Infrastructure/RouteNames.cs]: fixed

File: src/SFA.DAS.Roatp.CourseManagement.Domain/ApiModels/Provider.cs (hunk 3 above)
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement.Domain/ApiModels/Provider.cs]: removed

File: src/SFA.DAS.Roatp.CourseManagement.Web/Infrastructure/RouteNames.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement.Web/Infrastructure/RouteNames.cs]: done

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/ProviderController.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+        public ProviderController(IMediator mediator, ILogger<ProviderController> logger)
+        {
+            _mediator = mediator;
+            _logger = logger;
+        }
+
+        [Route("{ukprn}/provider-description", Name = RouteNames.GetProviderDescription)]
+        [HttpGet]
+        public async Task<IActionResult> ViewProductDescription(int ukprn)
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/ProviderController.cs:27]: `Product` Description ?

**Review by DevenKShah — APPROVED** [review]
There is minor spelling error in the controller.

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/ProviderController.cs (hunk 5 above)
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/ProviderController.cs:27]: lol what was I thinking.  Changed

**Review by DevenKShah — DISMISSED** [review]

**Review by DevenKShah — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**